src/compiler/transformers/esnext.ts

```diff
@@ -256,7 +260,7 @@ namespace ts {
                 visitEachChild(node.body, visitor, context);
             return setOriginalNode(
                 createFunctionExpression(
-                    /*modifiers*/ undefined,
+                    node.modifiers,

```

Comment:
Hmm. Good point. I'll confirm that they don't work and put them in a separate PR I think.